scripts/mainmenu.js

```diff
@@ -163,18 +243,30 @@ var MainMenuController = (function () {
 		$.persistentStorage.setItem(storedVar, !($.persistentStorage.getItem(storedVar) ?? false));
 	}
 
-	function _hideQuitMenu(bool = true) {
-		const panel = $("#QuitMenu");
-		if(!bool){
-			panel.RemoveClass("Disabled");
-			panel.AddClass("Enabled");
-			panel.RemoveClass("MainMenuSubContainerDisabled");
-			panel.AddClass("MainMenuSubContainerEnabled");
+
+	function _hideElement(element, bool = true) {
+		if(element != null){
+			const panel = $(element);
+			if(!bool){
+				panel.RemoveClass("hide");
+				panel.AddClass("show");
+				panel.RemoveClass("MainMenuFadeDisabled");
+				panel.AddClass("MainMenuFadeEnabled");
+				panel.visible = true;
+				_enableMainMenu(false);
+			} else {
+				$.Schedule(0.2, ()=>{
+					panel.RemoveClass("show")
+					panel.AddClass("hide")
+					panel.visible = false;
+				});
+				_enableMainMenu(true);
+				panel.RemoveClass("MainMenuFadeEnabled");
+				panel.AddClass("MainMenuFadeDisabled");
+				
+			}	
 		} else {
-			$.Schedule(0.2, ()=>panel.RemoveClass("Enabled"));
-			$.Schedule(0.2, ()=>panel.AddClass("Disabled"));
-			panel.RemoveClass("MainMenuSubContainerEnabled");
-			panel.AddClass("MainMenuSubContainerDisabled");
+			$.Warn("Unable to hide element: Undefined")
```

Comment:
nit: technically it's null right? i thought in js null and undefined were different

Reply:
In JS, there is "technically" a difference, but generally, they're the same thing, Null, means something that doesn't exist, there is no result, it's fully missing, undefined means there is a "reference" (like in this case, the variable) but it holds nothing, it's an undefined variable. it gets fucky with the way JS references it. both would work, undefined is more descriptive than null.